docs: move payment mechanism docs out of DEVELOP (#1134)
interledger/web-monetization-extension: docs/PAYMENTS.md

   1. Optionally, cancel the incoming payment grant, as we no longer need it (we already have the incoming payment). This prevents users wallet seeing _dangling_ grants in their wallet.
1. Find the `minSendAmount`:
   1. Find a possible amount for the receiving wallet using various heuristics (currency exchange rate, asset scales, asset codes etc.)
   2. Exponential probing: Try creating a quote with this amount until it doesn't fail with a "non-positive receive amount" error (i.e. the receiver has to receive at least one unit for a quote to succeed), increasing it in an exponential manner.
      - If the OpenPayments request with a "non-positive receive amount" error, error and includes a `minSendAmount` in error details (this is a relatively recent OpenPayments feature), stop the process to find the minimum sendable amount here.
   3. Binary search: Once a sendable amount is found, use binary search (between the sendable amount and previously attempted amount) to find the minimum sendable amount.
1. Create outgoing payment:
   1. Call the create outgoing payment OpenPayments API to create a fix-send outgoing payment. Include the following details:

>       - `accessToken`: The access token for the outgoing payment grant (i.e. the one we get from connecting the wallet). This token needs to be rotated once in a wIfIf the token is expired, try the payment again with a refreshed token.

Author: I'll stop using Apple Notes proofreading and rely on Sublime Text instead. 😆 
I'm surprised spell check didn't check it.

>       - `accessToken`: The access token for the outgoing payment grant (i.e. the one we get from connecting the wallet). This token needs to be rotated once in a wIfIf the token is expired, try the payment again with a refreshed token.

Author: ```suggestion
      - `accessToken`: The access token for the outgoing payment grant (i.e. the one we get from connecting the wallet). This token needs to be rotated once in a while: if the token is expired, retry the payment with a refreshed token.
```
